src/test/java/EthereumTest.java

```diff
@@ -0,0 +1,59 @@
+import org.bouncycastle.math.ec.custom.sec.SecP256K1Curve;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.web3j.crypto.Bip32ECKeyPair;
+import org.web3j.crypto.ECKeyPair;
+
+import java.util.HexFormat;
+
+public class EthereumTest {
+
+  /**
+   * Test data lifted from https://github.com/frequency-chain/frequency/blob/bac7782cba59247c7f89c0208720335a1289b8ce/js/ethereum-utils/test/signature.test.ts#L237
+   */
+  @Test
+  public void signTypedDataAddProvider() {
+    // GIVEN
+    final var privateKey = HexFormat.of().parseHex("5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133");
+    final var payload = new AddProviderPayload("12876327", new int[] {2, 4, 5, 6, 7, 8}, 100);
+
+    // WHEN
+    var signature = new byte[0];
+    try {
+      signature = Ethereum.signTypedData(payload, privateKey);
+    } catch (final Exception e) {
```

Comment:
I'd ditch the catch but if it's needed for some reason you shouldn't wipe the stack trace by only showing the message the stack trace is the best part. I imagine the `Assertions` library can handle a `Throwable` or at least I hope it can